react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({
+      email: e.target.value
+    });
+  }
+
+  handleSubmit(e) {
+    event.preventDefault();
+    console.log('state:', this.state);
+  }
+
+  render() {
+    return (
+      <form>
+        <label htmlFor="email">Email</label>
+        <input type="email" name="email" onChange={this.handleChange} value={this.state.email} />
+        <button type="submit" onClick={this.handleSubmit}>Submit</button>
```

Comment:
It's best to `handleSubmit` on the `<form>`, since buttons do not fire a `'submit'` event.